### PR TITLE
Snippets support in Annex H

### PR DESCRIPTION
**Pull request by silverhook**
As discussed in https://github.com/spdx/spdx-spec/pull/464 this PR expands Annex H to include snippet-level tags in source code.

PR instead of https://github.com/spdx/spdx-spec/pull/464

### CODE REVIEW

**Comment by silverhook** [comment]
I do not remember what the outcome on the calls was regarding how and if at all nesting of snippets should work, so in this version I completely left this part out.

AFAICR two options are possible:
1. nesting like brackets and in most markup where the closest begin and end tag belong to each other
2. no nesting, so before a new snippet begin tag is allowed, the previous snippet has to have an end tag

I’m more in favour of 1.

**Comment by goneall** [comment]
> I do not remember what the outcome on the calls was regarding how and if at all nesting of snippets should work, so in this version I completely left this part out.

I recall the consensus was option 1 - nesting like brackets and in most markup where the closest begin and end tag belong to each other

**Comment by goneall** [comment]
@silverhook - only comment is to mention the nesting (option 1 above).

File: chapters/file-tags.md [hunk 1]
```diff
@@ -50,10 +50,62 @@ Multiple-line values are not recommended, because doing so will make it harder f
 
 Version 3.0 of the [REUSE Software guidelines](https://reuse.software/spec/) implements this format, via a recommendation to use the tag `SPDX-FileCopyrightText:` to include copyright notices as part of a file's comment headers.
 
-## H.3 Caveats <a name="H.3"></a>
+## H.3 File Tags Within Snippets <a name="H.3"></a>
+
+If certain SPDX file tags are to apply only to a certain snippet instead of the whole file, the start of such snippet should be marked with `SPDX-SnippetBegin` and its end with `SPDX-SnippetBegin`.
```
Comment (zvr) [inline chapters/file-tags.md:55]: Ahem. Probably more useful to denote the end with `SPDX-SnippetEnd` instead with `Begin`.

File: chapters/file-tags.md [hunk 2]
```diff
@@ -50,10 +50,62 @@ Multiple-line values are not recommended, because doing so will make it harder f
...
+SPDX-SnippetBegin
+SPDX-License-Identifier: BSD-2-Clause
+SPDX-SnippetCopyrightText: Copyright Example Company
+
+...
+
+SPDX-SnippetEnd
+
+```
+
```
Comment (zvr) [inline chapters/file-tags.md:102]: Add something like:

Snippets may nest, and this is denoted by having `SPDX-SnippetBegin`/`SPDX-SnippetEnd` pairs within other pairs, in the same way that parentheses nest in mathematical expressions.
In the case of nested snippets, the SPDX file tags are considered to apply to the inner-most snippet.

**Review by zvr — CHANGES_REQUESTED** [review]

File: chapters/file-tags.md [hunk 3]
```diff
@@ -50,10 +50,62 @@ Multiple-line values are not recommended, because doing so will make it harder f
 
 Version 3.0 of the [REUSE Software guidelines](https://reuse.software/spec/) implements this format, via a recommendation to use the tag `SPDX-FileCopyrightText:` to include copyright notices as part of a file's comment headers.
 
-## H.3 Caveats <a name="H.3"></a>
+## H.3 File Tags Within Snippets <a name="H.3"></a>
```
[the remark this answers is not in the document]
Comment (silverhook) [reply chapters/file-tags.md:53]: Will change it, thanks.

But chapter titles still use title case, right? (That’s where I copied the style from)

[the remark this answers is not in the document]
Comment (zvr) [reply chapters/file-tags.md:53]: Eh, no: `8 File information section` or `3 Terms and definitions`.
It was one of the changes when we were preparing for ISO...
`

File: chapters/file-tags.md [hunk 4]
```diff
@@ -50,10 +50,62 @@ Multiple-line values are not recommended, because doing so will make it harder f
...
+
+If certain SPDX file tags are to apply only to a certain snippet instead of the whole file, the start of such snippet should be marked with `SPDX-SnippetBegin` and its end with `SPDX-SnippetBegin`.
+
+Any Snippet Information (Clause [9](snippet-information.md)) and short-form license identifiers (Annex [E](using-SPDX-short-identifiers-in-source-files.md)) tags found between begin and end tags mentioned above apply only to such snippet.
```
[the remark this answers is not in the document]
Comment (silverhook) [reply chapters/file-tags.md:57]: I didn’t want to mess around with the existing parts too much, but have tried to implement what you’re (reasonably!) asking for @zvr.

It feel we can simplify this by explicitly separating snippet tags from file tags. I’ll create a new PR for what I mean.

[the remark this answers is not in the document]
Comment (silverhook) [reply chapters/file-tags.md:57]: A restructured version, that IMHO looks better: https://github.com/spdx/spdx-spec/pull/719

**Comment by silverhook** [comment]
As an alternative to this PR, I created also https://github.com/spdx/spdx-spec/pull/719, which IMHO is better structured.

I am keeping both open so we can decide between:
- the clearer and better structured https://github.com/spdx/spdx-spec/pull/719; and
- keeping the original structure in https://github.com/spdx/spdx-spec/pull/718

File: chapters/file-tags.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (silverhook) [reply chapters/file-tags.md]: So we should change the chapter title as well then?

**Comment by silverhook** [comment]
Closing in favour of the approved #719

**Pull request closed without merging**